components/supervisor/pkg/supervisor/supervisor.go

```diff
@@ -220,13 +229,17 @@ func Run(options ...RunOption) {
 	wg.Add(1)
 	go startSSHServer(ctx, cfg, &wg)
 	wg.Add(1)
-	go taskManager.Run(ctx, &wg)
+	tasksSuccessChan := make(chan bool, 1)
+	go taskManager.Run(ctx, &wg, tasksSuccessChan)
 	wg.Add(1)
 	go socketActivationForDocker(ctx, &wg, termMux)
 
 	if !cfg.isHeadless() {
 		wg.Add(1)
 		go portMgmt.Run(ctx, &wg)
+	} else {
```

Comment:
nit: Now that we have an else branch the negation in the condition feels awkward

Reply:
nit: you suggested it exactly this way a few iterations back :-) But will change...